### PR TITLE
Parachain v0.9.7

### PR DESCRIPTION
**Pull request by zqhxuyuan**
phase 1:
1. merge from rococo-v1 to master
2. upgrade all components version to v0.9.4
3. fix some upgrading compatiable api

phase 2:
1. upgrade to v0.9.7
2. parachain use aura consens

### CODE REVIEW

File: bin/node-prep/cli/src/chain_spec/jupiter.rs [hunk 1]
```diff
@@ -424,18 +424,21 @@ fn testnet_genesis(
                 })
                 .collect::<Vec<_>>(),
         },
-        pallet_democracy: Default::default(),
-        pallet_collective_Instance1: CouncilConfig {
+        // pallet_democracy: Default::default(),
```
Comment (atenjin) [inline bin/node-prep/cli/src/chain_spec/jupiter.rs]: 这些没用的注释都删了好了

File: bin/node/cli/src/command.rs [hunk 2]
```diff
@@ -47,36 +50,36 @@ impl SubstrateCli for Cli {
...
+        "https://github.com/patractlabs/jupiter".into()
     }
 
     fn copyright_start_year() -> i32 {
         2020
     }
 
     fn load_spec(&self, id: &str) -> std::result::Result<Box<dyn sc_service::ChainSpec>, String> {
-        load_spec(id, self.run.parachain_id.unwrap_or(200).into())
+        load_spec(id, self.run.parachain_id.unwrap_or(24).into())
```
Comment (atenjin) [inline bin/node/cli/src/command.rs]: parachain id 重新指定好

File: bin/node/cli/src/command.rs [hunk 3]
```diff
@@ -47,36 +50,36 @@ impl SubstrateCli for Cli {
...
         env!("SUBSTRATE_CLI_IMPL_VERSION").into()
     }
 
     fn description() -> String {
-        "Parachain Collator Template\n\nThe command-line arguments provided first will be \
+        "Patract parachain collator\n\nThe command-line arguments provided first will be \
 		passed to the parachain node, while the arguments provided after -- will be passed \
 		to the relaychain node.\n\n\
-		parachain-collator [parachain-args] -- [relaychain-args]"
+		patract [parachain-args] -- [relaychain-args]"
```
Comment (atenjin) [inline bin/node/cli/src/command.rs]: 改成 jupiter

File: bin/node/cli/src/command.rs [hunk 4]
```diff
@@ -47,36 +50,36 @@ impl SubstrateCli for Cli {
...
+        "Patract Parachain Collator".into()
     }
 
     fn impl_version() -> String {
         env!("SUBSTRATE_CLI_IMPL_VERSION").into()
     }
 
     fn description() -> String {
-        "Parachain Collator Template\n\nThe command-line arguments provided first will be \
+        "Patract parachain collator\n\nThe command-line arguments provided first will be \
```
Comment (atenjin) [inline bin/node/cli/src/command.rs]: Patract Jupiter parachain collator

File: bin/node/cli/src/command.rs [hunk 5]
```diff
@@ -268,19 +286,17 @@ pub fn run() -> Result<()> {
...
-                    config.base_path.as_ref().map(|x| x.path().join("polkadot")),
-                    relay_chain_id,
+                    &config,
                     [RelayChainCli::executable_name().to_string()]
                         .iter()
                         .chain(cli.relaychain_args.iter()),
                 );
 
-                let id = ParaId::from(cli.run.parachain_id.or(para_id).unwrap_or(100));
+                let id = ParaId::from(cli.run.parachain_id.or(para_id).unwrap_or(1));
```
Comment (atenjin) [inline bin/node/cli/src/command.rs:298]: 这里的id1 统一一下和上面一样

File: primitives/io/src/lib.rs [hunk 6]
```diff
@@ -19,87 +19,88 @@ pub trait Pairing {
         false
     }
 
-    fn call(func_id: u32, input: &[u8]) -> Option<Vec<u8>> {
-        curve::call(func_id, input).ok()
+    fn call(_func_id: u32, _input: &[u8]) -> Option<Vec<u8>> {
+        // curve::call(func_id, input).ok()
```
Comment (atenjin) [inline primitives/io/src/lib.rs:23]: 这些尝试恢复下，看能不能编译通过

**Comment by zqhxuyuan** [comment]
close as v0.9.8 comeup

**Pull request closed without merging**